libdnf/comps/environment/query.cpp

```diff
@@ -63,25 +65,29 @@ EnvironmentQuery::EnvironmentQuery(const BaseWeakPtr & base) : base(base) {
             continue;
         }
 
+        repoid = solvable->repo->name;
+
         // Add installed environments directly, because there is only one solvable for each
-        if (strcmp(solvable->repo->name, "@System") == 0) {
+        if (strcmp(repoid, "@System") == 0) {
             Environment environment(base);
             environment.environment_ids.push_back(EnvironmentId(solvable_id));
             add(environment);
         } else {
             // Create map of available environments:
             // for each environmentid (SOLVABLE_NAME), list all corresponding solvable_ids
-            // TODO(pkratoch): Sort solvable_ids for each environmentid according to repo id
             available_map[solvable_name_pair.second].insert(
-                available_map[solvable_name_pair.second].begin(), solvable_id);
+                available_map[solvable_name_pair.second].end(), std::pair<Id, std::string_view>(solvable_id, repoid));
         }
     }
 
     // Create environments based on the available_map
-    for (const auto & item : available_map) {
+    for (auto item : available_map) {
```

Comment:
If you don't use a reference, this makes a copy of each item in the map. `auto & item` is much more effective (if you don't mind modifying the data in the map).